docs/DONOTMERGE/pr-402.md

```diff
@@ -0,0 +1,108 @@
+# How to provide ledger events to client apps
+
+## Introduction
+
+On Thursday Oct 12, we discussed [PR #402](https://github.com/input-output-hk/ouroboros-consensus/pull/402) during the Consensus Office Hours.
+@KtorZ @koslambrou @kderme @nfrisby @amesgen did most of the talking.
+UTxO HD came up, so @jasagredo likely also would have been in that list if he hadn't been on a bank holiday.
+
+This document captures the key points of the discussion.
+
+## Executive summary
+
+- The Consensus Team considers this PR itself innocuous, because we don't anticipate it being difficult to review or to maintain or to design around (famous last words?).
+- However, there is one axis along which we think it is under-specified, and so our above judgement would be revisited after _the exact necessary semantics_ of `rnHandleLedgerEvent`  are better identified.
+- During the call we discussed some possible semantics of `rnHandleLedgerEvent`.
+  The "simple" one (which _we think_ this PR currently satisfies) would be easy for Consensus to maintain, but offloads a significant amount of complexity to the ultimate client.
+  The "complex" one relieves the clients of that burden, but would require a significantly larger PR here.
+- We are concerned that the "simple" semantics would let us merge this PR, but then _eventually_ no one would actually use the `rnHandleLedgerEvent` interface because the required client complexity is too hard.
+  We are hesitant to expand the Consensus interface (even with the "simple" semantics) if there's a clearly-plausible path to it becoming dead code in the near-future (there's only so much dev power available, it should ideally be allocated to long-lived aka "long-term" solutions).
+
+## Requirements
+
+Any complete solution for the needs of the relevant client applications _must_ handle all three of the following tasks.
+
+- staying synced with a node when its selection gains some blocks
+- staying synced with a node when its selection loses some blocks, aka rollbacks
+- becoming synced with a node
+
+The primary desiderata are as follows.
+
+- The client can be implemented in any language.
+- The client has access to the per-block ledger events, or at least the ones known to be desirable (eg calculated rewards and calculated stake distributions).
+- The client need not hold its own copy of the ledger state in memory.
```

Comment:
Maybe we should already state here to what extent this is addressed by UTxO HD (probably just that is is *partially* addressed).